### PR TITLE
DRUP-1051: adds Open Now indicator in sidebar via ajax

### PR DESCRIPTION
**Pull request by z3cka**
@akohler & @emb03 please take a look. I'd like to deploy this to `www-test` as soon as possible so I can get this to `www-stage` for load testing on Friday.

### CODE REVIEW

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js [hunk 1]
```diff
@@ -73,6 +73,29 @@ var libHoursApp = libHoursApp || {};
         var view = new libHoursApp.appView({el: '.'+element.classList[1]});
         Drupal.libHoursAppViews.push(view);
       });
+
+      // Open Now backbone stuff
+      var libHoursAppLocationModel = Backbone.Model.extend({
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: Is there any reason to separate the new "today" hours js functions into a separate file, called only when relevant?   Though probably a tiny drop in the ocean of JS loaded on every page...

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js [hunk 2]
```diff
@@ -73,6 +73,29 @@ var libHoursApp = libHoursApp || {};
...
+        url: '/ajax/libhours/hours/today/'+ Drupal.settings.ucla_libhours.hours.lid
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: The `lid` value is defined only for top-level locations, not for individual destinations, correct (which seem to use 0 if called directly, like this Biomed Reference Services destination https://webservices.library.ucla.edu/calendar/hours/today/3289)

But I think it doesn't matter since the API still returns `times.currently_open` for destinations too.

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module [hunk 3]
```diff
@@ -11,6 +11,13 @@ function ucla_libhours_menu() {
     'page callback' => 'ucla_libhours_get_hours',
     'page arguments' => array(2, 4),
   );
+  # open now endpoint
+  $items['ajax/libhours/hours/today/%'] = array(
+    'title' => 'libhours open now',
+    'access callback' => true,
+    'page callback' => 'ucla_libhours_get_open_now',
+    'page arguments' => array(4),
```
Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: What's `array(4)` - is this passing an empty array?  Drupal magic we don't care about?  Consider adding a comment to the code.

Comment (z3cka) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module]: The magic comes from the Drupal's `hook_menu`'s API on how to use [Wildcards in Paths](https://api.drupal.org/api/drupal/modules%21system%21system.api.php/function/hook_menu/7.x#sec_path_wildcards); `array(4)` simply grabs the 4th argument in the path of `ajax/libhours/hours/today/%` and passes it through to the `'page callback'` function.

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js (hunk 1 above)
Comment (z3cka) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: Maybe, given that this code is only run on the location pages... and the other table code get used in other places, but this addition is small enough that adding it to another file to load may offset any benefit we get otherwise. 

I will think about it some more. I assume this is not a blocker.

Comment (akohler) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: Definitely not a blocker.  Having utility code all in one file is fine.  My question's more about javascript getting loaded everywhere if if's only needed in specific places.  I agree it may cost more overall to split into separate files and manage those appropriately.  No change needed, just something I was wondering about.

Comment (z3cka) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: nah, not for this case... we are not minifying and optimizing our js at this time... doing that would negate moving this to its own file anyways. 

I would like to do an entire front end performance assessment and fix  issues like this.

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js (hunk 2 above)
Comment (z3cka) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: No, all locations/destinations get `lid`'s in libcal. Correct, the `lid` property returns as `0` in the json when a location is requested directly. 

`Drupal.settings.ucla_libhours.hours.lid` is a value passed from drupal's php to drupal's js via: https://github.com/z3cka/libraryweb-site/blob/4639c8943ba07762481cdcaf30dbcaaa9962371e/www/sites/all/modules/custom/ucla_libhours/ucla_libhours.module#L130-L132
This is a part of the `#attached` property.

So in this context, `Drupal.settings.ucla_libhours.hours.lid` always has a value.

**Comment by z3cka** [comment]
deploying to `www-test` now

File: www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js [hunk 4]
```diff
@@ -73,6 +73,29 @@ var libHoursApp = libHoursApp || {};
...
+        el: '.open-now-pane',
+        initialize: function() {
+          this.model = new libHoursAppLocationModel();
+          this.listenTo(this.model, 'sync', this.render);
+          this.model.fetch();
+        },
+        render: function() {
+          var todays_hours = this.model.toJSON();
+          if (todays_hours.locations["0"].times.currently_open) {
+            $(this.el).html('<span class="location-open-now">Open</span>');
```
Comment (z3cka) [inline www/sites/all/modules/custom/ucla_libhours/ucla_libhours.js]: Oops, this should be "Open Now"... will make another commit+deploy in the morning.
